clippy_lints/src/bool_to_int_with_if.rs

```diff
@@ -0,0 +1,123 @@
+use rustc_ast::{ExprPrecedence, LitKind};
+use rustc_hir::{Block, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+use clippy_utils::{diagnostics::span_lint_and_then, source::snippet_block};
+use rustc_errors::Applicability;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Instead of using an if statement to convert a bool to an int, this lint suggests using a `as` coercion or a `from()` function.
+    /// ### Why is this bad?
+    /// Coercion or `from()` is idiomatic way to convert bool to a number.
+    /// Both methods are guaranteed to return 1 for true, and 0 for false. See https://doc.rust-lang.org/std/primitive.bool.html#impl-From%3Cbool%3E
+    ///
+    /// ### Example
+    /// ```rust
+    /// let condition = false;
+    /// if condition {
+    ///     1_i64
+    /// } else {
+    ///     0
+    /// };
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// let condition = false;
+    /// i64::from(condition);
+    /// ```
+    /// or
+    /// ```rust
+    /// let condition = false;
+    /// condition as i64;
+    /// ```
+    #[clippy::version = "1.64.0"]
+    pub BOOL_TO_INT_WITH_IF,
+    style,
+    "using if to convert bool to int"
+}
+declare_lint_pass!(BoolToIntWithIf => [BOOL_TO_INT_WITH_IF]);
+
+impl<'tcx> LateLintPass<'tcx> for BoolToIntWithIf {
+    fn check_expr(&mut self, ctx: &LateContext<'tcx>, expr: &'tcx rustc_hir::Expr<'tcx>) {
+        if !expr.span.from_expansion() {
+            check_if_else(ctx, expr);
+        }
+    }
+}
+
+fn check_if_else<'tcx>(ctx: &LateContext<'tcx>, expr: &'tcx rustc_hir::Expr<'tcx>) {
+    if_chain!(
```

Comment:
NIT: If chains are usually used with curly brackets. Alternatively, you can also use rust's if let chains. They cover the same functionality and allow automatic formatting.

See: https://rust-lang.github.io/rfcs/2497-if-let-chains.html

Reply:
Same goes for other `if_chains!` macros,  I just didn't add any comments for them :upside_down_face:

Reply:
fixed in [d857ce3](https://github.com/rust-lang/rust-clippy/pull/9086/commits/d857ce3109209ec6c16a9515821675a7e1cd211e).

I changed all `if_chain!` to rust's if chains.

Though formatting still doesn't seem to work. It may be an issue with rustfmt. After looking through open issues in its repo I didn't find any relating to `if let` chains specifically.

And I feel that new version looks kind of ugly. I won't resolve this for now. Let me know if I can improve formatting of updated code.

On a side note - I've been resolving conversations about suggestions I've fixed. Is it okay or should I leave resolving convos to maintainers?

Reply:
Thank you! The formatting of rustc's if let chains is still open somewhere AFAIK, the advantage Clippy gains from them, is that code inside the `then` and `else` block will be formatted, which wasn't before. I'll have a look at it in the next few days.

> I've been resolving conversations about suggestions I've fixed. Is it okay or should I leave resolving convos to maintainers?

This is different for projects and maintainers. Personally, I (as a reviewer) prefer resolving them myself when I've checked that they've been solved. Simple suggestions can be resolved by either party. But that is just a preference, either way is fine with me. :upside_down_face: